test/test.js

```diff
@@ -2922,6 +2922,14 @@ describe('Page', function() {
       expect(await page.evaluate(() => __injected)).toBe(42);
     });
 
+    it('should work with a url and type=module', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+      const scriptHandle = await page.addScriptTag({ url: '/injectedfile.js', type: 'module' });
```

Comment:
can we check that `import` statement works in the injected url? e.g. via importing some script that yields `console.log`.

Reply:
Yes, I started with that. But then eslint was failing because you must put import at the top (so I couldn't use it inside my test, in an evaluate). 
And then I thought: I don't want to `test type="module"` itself, I assume it will work, as long as I add type="module" to my script. WDYT?

Reply:
One more point - user can add more kind of types (I fixed that). So this feature is not only about `type=module` (:

Reply:
> And then I thought: I don't want to test type="module" itself, I assume it will works, as long as I add type="module" to my script. WDYT?

Well, there's a claim in #2078 that `type="module"` doesn't work this way, so a test would be a good sanity check.

> But then eslint was failing because you must put import at the top (so I couldn't use it inside my test, in an evaluate statement).

Ah, you can create a test asset `assets/es6import.js` that uses modules. This should make eslint happy.

Reply:
yes, so after playing with `type="module"`, it seems that es6 modules with `type="module"` and chromium doesn't work. I wrote a test which failed for not recognizing es6 syntax. I then tried running [this](https://stackoverflow.com/a/45745913/916450) code snippet on Canary / Chromium and got the same error.

Reply:
@yanivefraim that's weird. I'm pretty sure chromium supports es6 modules: https://caniuse.com/#feat=es6-module

Do you have time to debug this further?

Reply:
Modules definitely work. I'm using them in several projects:)

@yanivefraim can you share the test case? We can take over this PR if you don't have time. There use case is a neat one (sharing code with node). I can see other cases like universal js with pptr running on a server and injecting modules into a page at serve time.

Reply:
@aslushnikov @ebidel  - I will love to give it a try... (didn't have much time last two weeks but now I will find time. if this is ok with you)

Reply:
@ebidel @aslushnikov - so apparently It was my mistake (this is what happens when working on PRs at 2am...). I forgot to add `.js` in my import (: 
Fixed my test to actually be an e2e and test that modules are working.